### PR TITLE
[Duplicate of #6816] - Optimizing ML Models with Optuna

### LINKED ISSUE #5749: Optimizing ML Models with Optuna
### Topic Suggestion
Optimizing ML Models with Optuna
#### Proposed article introduction
Optuna is the SOTA algorithm used to make changes to ML and deep learning models. It all depends on the Bayesian fine-tuning method. It stops trying combinations that don't improve our score any more and only tries combinations that do.

People who work on ML models have a hard time. In order to make it easier for us to fine-tune our model, there are a lot of tools out there. You need to make sure that our data is important and that our ML model fits on it perfectly. Here, we'll learn how to make an Xgboost regressor work even better.

When you use machine learning, a hyperparameter is a parameter that can be changed to change how the learning process works.  The values of hyperparameters can't be guessed from the data. It's important to set these parameters so that the model can learn from our data. You can change the learning parameters so that the model fits our data and learns from it to make smart predictions.
#### Key takeaways 
- Visualization of Tuning
- Comparing the Tuned Model and Base Model
- Creating Hyper-Parameters Space

#### Article quality
In this article, we'll learn how to improve ML models and how to find the best parameters for the models we make. The code will be easy to read and understand.
#### References
N/A
### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by NdoloBonface**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #5749 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by NdoloBonface** [comment]
@srishilesh

**Comment by NdoloBonface** [comment]
Duplicate of #6816

**Comment by NdoloBonface** [comment]
@LinusMuema @srishilesh is it not a tech tutorial?

**Comment by LinusMuema** [comment]
It is...

Just that there were many labels so one can get confused at which point the review is at.

**Comment by hectorkambow** [comment]
image

**Review by srishilesh — APPROVED** [review]
Hey @NdoloBonface ,

Great article. Thanks for making the requested changes.

I've made some major changes to the formatting and flow of the article. Please go through my commits carefully to understand them.

Ensure you see them before you start writing your next article.

Moving for the final review.

Thanks.
